arcsi/api/utils.py

```diff
@@ -158,7 +159,45 @@ def get_shows():
             )
     return shows
 
-def item_duplications_number(item):
-    name_occurrence = int(db.session.query(db.func.count()).filter(Item.name == item.name, Item.number == item.number).scalar())
+def show_item_duplications_number(item):
+    existing_items = Show.query.filter(Show.id == item.shows[0].id).first().items
+    existing_items_with_same_name = existing_items.filter_by(name=item.name)
+    name_occurrence = existing_items_with_same_name.count()
     app.logger.error("Name_occurence (duplicate detection): {}".format(name_occurrence))
     return name_occurrence
+
+def comma_separated_params_to_list(param):
+    result = []
+    for val in param.split(','):
+        if val:
+            result.append(val)
+    return result
+
+def filter_show_items(show, items, archived, latest):
+    items = sorted(items, key=lambda x: x["play_date"], reverse=True)
+    if archived:
+        already_aired_items = [
+            show_item
+            for show_item in items
+            if (show_item.get("archived") &
+                (datetime.strptime(show_item.get("play_date"), "%Y-%m-%d") + timedelta(days=1) < datetime.today()))
+        ]
+        return get_show_items(show, already_aired_items, latest)
+    else:
+        return get_show_items(show, items, latest)
+        
+def get_show_items(show, items, latest):
+    do = DoArchive()
+    if latest:
+        items = items[0]
+        items["image_url"] = do.download(show.archive_lahmastore_base_url, items["image_url"])
+        items["name_slug"] = normalise(items["name"])
+        return items
+    else:
+        items = items
```

Comment:
This may be not necessary you can just loop through the incoming argument of `items` and return the modified version of it.

Reply:
true 😅